webcheck.go

```diff
@@ -188,23 +188,49 @@ func (fm *Frontman) runWebCheck(data WebCheckData) (m map[string]interface{}, er
 		}
 
 	} else {
-		b := make([]byte, 512)
-		for true {
-			n, err := resp.Body.Read(b)
-			totalBytes += n
-			if err != nil {
-				break
-			}
+		_, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			err = fmt.Errorf("got error while reading response body: %s", err.Error())
 		}
-
 	}
 
-	m[prefix+"bytesReceived"] = totalBytes
+	bytesReceived := bodyReaderWithCounter.Count()
+
+	m[prefix+"bytesReceived"] = bytesReceived
 	m[prefix+"totalTimeSpent_s"] = time.Since(startedConnectonAt).Seconds()
 	seconds := time.Since(wroteRequestAt).Seconds()
 	if seconds > 0 {
-		m[prefix+"downloadPerformance_bps"] = int64(float64(totalBytes) / seconds) // Measure download speed since the request sent
+		m[prefix+"downloadPerformance_bps"] = int64(float64(bytesReceived) / seconds) // Measure download speed since the request sent
 	}
 
 	return
 }
+
+// gzipReader wraps a response body so it can lazily
+// call gzip.NewReader on the first call to Read
+type gzipReader struct {
```

Comment:
I would put this into an extra file or package. This can probably be reused and is not directly coupled to webchecks.

Reply:
done